tests/unit/src/test/scala/tests/NewFileTemplateSuite.scala

```diff
@@ -0,0 +1,33 @@
+package tests
+
+import munit.ScalaCheckSuite
+import scala.meta.internal.metals.NewFileTemplate
+import org.scalacheck.Gen
+import org.scalacheck.Prop.forAll
+
+class NewFileTemplateSuite extends BaseSuite with ScalaCheckSuite {
+
+  test("cursor-marker-errors") {
+    intercept[IllegalArgumentException] {
+      NewFileTemplate("no cursor markers")
+    }
+    intercept[IllegalArgumentException] {
+      NewFileTemplate("many cursor @@ markers @@")
+    }
+  }
+
+  property("cursor-marker-position") {
+    val template =
+      s"""|package a
+          |
+          |case class Foo()
+          |""".stripMargin
+    val cursorOffsetGen = Gen.chooseNum(0, template.length)
+    forAll(cursorOffsetGen) { cursorOffset =>
```

Comment:
What is the difference between `forAll` here and `0.to(template.length).foreach { .. }`?

Reply:
ah, good point: in this case just the number of iterations (100 with ScalaCheck). I can revert to just using a range

Reply:
I think `1.to(100)` would be clearer and more consistent with the rest of the tests. Property based testing is more helpful when you can't tests all possible inputs.